### DISCUSSION TITLE
Battery Pack Design Automation (Project 142)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#13] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project #142](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/tree/main/projects/Battery%20Pack%20Design%20Automation)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by ghost** [comment]
Does this require bachelor-level electronic engineering expertise? (CC @kgrand-mw)

**Reply by kgrand-mw** [reply]
No it does not.  A strong mathematic and optim background would be helpful.

**Reply by ghost** [reply]
> https://github.com/mathworks/Battery-Pack-Model-Simscape

Is this the project itself, a generic but similar thing, or mostly unrelated?

**Reply by kgrand-mw** [reply]
Similar but different.  There is an optimal solution to battery pack sizing which is the goal of our project.

**Comment by lhcalpha** [comment]
Hello ,I am Parth from IIT Roorkee(India), I am new to battery design but interested in this project. I read about developing the battery model so in my understanding to get the SOC input in the battery model we need to first fit the data(from generate parameter data) for getting the parameters of the equivalent circuit battery, is it the right way to think?

Any help will be beneficial

**Reply by kgrand-mw** [reply]
Hello Parth, we can setup some time to chat if this would be helpful.

**Reply by lhcalpha** [reply]
Yes sure, what time will you prefer ?

**Comment by lhcalpha** [comment]
Hello @robertogl @kgrand-mw ,I am a bit confused on giving SOC as input because SOC is calculated from the output of the model, a bit help will be good.

**Reply by kgrand-mw** [reply]
Hello Parth,  the question is unclear?  SOC is an output that is a function of accumulated battery current.

**Reply by lhcalpha** [reply]
Yes ,that's my doubt because when I read some papers that mentioned SOC will be one of the outputs but in the problem statement it is written to take input as temperature and SOC.
I might be unable to get the main essence so a bit help will be good

**Comment by kgrand-mw** [comment]
I am in the eastern time zone – tomorrow at 8AM EST should work.

Microsoft Teams meeting
Join on your computer or mobile app
Click here to join the meeting<https://teams.microsoft.com/l/meetup-join/19%3ameeting_OGU4NTkwNjMtNjM4YS00YWJhLTk2OTAtMjZkMmJiY2U4ZTJk%40thread.v2/0?context=%7b%22Tid%22%3a%2299dd3a11-4348-4468-9bdd-e5072b1dc1e6%22%2c%22Oid%22%3a%228944bf9a-81e5-430a-a755-7fd5465a3c78%22%7d>

**Reply by lhcalpha** [reply]
Okay thanks a lot, I will join surely at that time it is 5:30 PM according to IST

**Reply by lhcalpha** [reply]
Thanks for the meet and valuable resources kerry, they will help me

**Reply by lhcalpha** [reply]
Apologies @kgrand-mw was off pace for some days I had a doubt, about the cube consideration of a module, suppose there are n cells in a module the number n will be known after optimization so, what I understand is that we need to assume a cube module in which there are these cells  or we have to assume the cells as cube for thermal transmission consideration

**Comment by kgrand-mw** [comment]
Please no need to apologize.  I would start the problem with the assumption that the cells will occupy a cubic space with thermal conductivity on all six slide that may have different/unique thermal coefficients.  This should allow an assessment of the overall thermal load, mass increase, colling requirements of the overall system at a high level and at some point the addition of battery cells will produce negative returns (less range and we know that cost will always increase with more cell additions).  Once we understand this the next step would be for module packing and module thermal design considerations.

**Reply by lhcalpha** [reply]
Okay thanks, sir so I am going to improve the thermal model of the cell and assume that my n cells will occupy cubic space, just a small doubt that each and every cell will have some differing temperature even in a single cube module, so that must be considered while doing thermal considerations in the module on all six sides as inside we have n cells with different temperature outputs. Right ? 
I might be dumb a bit to understand

Thanks

**Comment by kgrand-mw** [comment]
I think you are correct.  You are approximating the flow of thermal energy on six sides of a cube and the cube may connect to another cube or a cooling plate or ambient temperature.

**Reply by lhcalpha** [reply]
Okay sir thanks, I will move forward with the assumption of battery cell as a cube and then packing those cube cells in a module for optimizing temperature flow through module, and then packing those modules in a battery pack

**Reply by lhcalpha** [reply]
Hello sir, so I understand now how to assume the cells just a small clarification, face of cube that will be touching each other what type of boundary condition should be assumed ? For example in the problem statement it is mentioned for taking convection into consideration, but do we have to consider conduction too at the joining faces (it will be complex in my opinion).
A bit of help will guide me in right direction I guess

**Reply by lhcalpha** [reply]
Hello Sir @kgrand-mw , Sorry I am off pace since many days my exams are going on currently, they are ending soon, can we have a small meet if you don't have any problem? I have some doubts, so that I can move a bit more into the project

**Reply by Arun-Mohapatra** [reply]
Hello @lhcalpha 
Are you still working on the project?
I would like to join and contribute.

**Comment by BhargavaVenkataSiva** [comment]
Hi, 
Is this discussion still online?

**Reply by robertogl** [reply]
Yep.

**Reply by BhargavaVenkataSiva** [reply]
Hi, 
I am new to these types of projects and I have no idea where to start. There is no explanation whatsoever, right now I am doing the literature research but after that I do not have any idea to start.
